tests/subsys/net/lib/downloader/src/main.c

```diff
@@ -0,0 +1,1359 @@
+
+	evt = dlc_wait_for_event(DOWNLOADER_EVT_ERROR, K_SECONDS(3));
+
+	downloader_deinit(&dlc);
+	dlc_wait_for_event(DOWNLOADER_EVT_DEINITIALIZED, K_SECONDS(1));
+}
+
+void test_downloader_get_default_proto_http(void)
+{
+	int err;
+	struct downloader_evt evt;
+
+	err = downloader_init(&dlc, &dlc_config_cb_abort);
+	TEST_ASSERT_EQUAL(0, err);
+
+	zsock_getaddrinfo_fake.custom_fake = zsock_getaddrinfo_server_ipv6_fail_ipv4_ok;
+	zsock_freeaddrinfo_fake.custom_fake = zsock_freeaddrinfo_server_ipv4;
+	z_impl_zsock_socket_fake.custom_fake = z_impl_zsock_socket_http_ipv4_ok;
+	z_impl_zsock_connect_fake.custom_fake = z_impl_zsock_connect_ipv4_ok;
+	z_impl_zsock_setsockopt_fake.custom_fake = z_impl_zsock_setsockopt_http_ok;
+	z_impl_zsock_sendto_fake.custom_fake = z_impl_zsock_sendto_ok;
+	z_impl_zsock_recvfrom_fake.custom_fake = z_impl_zsock_recvfrom_http_header_then_data;
+
+	/* Default to http when no proto is specified. */
+	err = downloader_get(&dlc, &dlc_host_cfg, NO_PROTO_URI, 0);
+	TEST_ASSERT_EQUAL(0, err);
+
+	evt = dlc_wait_for_event(DOWNLOADER_EVT_DONE, K_SECONDS(3));
+
+	downloader_deinit(&dlc);
+	dlc_wait_for_event(DOWNLOADER_EVT_DEINITIALIZED, K_SECONDS(1));
+}
+
+void test_downloader_get_default_proto_https(void)
+{
+	int err;
+	struct downloader_evt evt;
+
+	err = downloader_init(&dlc, &dlc_config_cb_abort);
+	TEST_ASSERT_EQUAL(0, err);
+
+	zsock_getaddrinfo_fake.custom_fake = zsock_getaddrinfo_server_ok;
+	zsock_freeaddrinfo_fake.custom_fake = zsock_freeaddrinfo_server_ipv6;
+	z_impl_zsock_socket_fake.custom_fake = z_impl_zsock_socket_https_ipv6_ok;
+	z_impl_zsock_connect_fake.custom_fake = z_impl_zsock_connect_ipv6_ok;
+	z_impl_zsock_setsockopt_fake.custom_fake = z_impl_zsock_setsockopt_https_ok;
+	z_impl_zsock_sendto_fake.custom_fake = z_impl_zsock_sendto_ok;
+	z_impl_zsock_recvfrom_fake.custom_fake = z_impl_zsock_recvfrom_http_header_then_data;
+
+	/* Default to http when no proto is specified. */
+	err = downloader_get(&dlc, &dlc_host_cfg_w_sec_tags, NO_PROTO_URI, 0);
+	TEST_ASSERT_EQUAL(0, err);
+
+	evt = dlc_wait_for_event(DOWNLOADER_EVT_DONE, K_SECONDS(3));
+
+	downloader_deinit(&dlc);
+	dlc_wait_for_event(DOWNLOADER_EVT_DEINITIALIZED, K_SECONDS(1));
+}
+
+void test_downloader_get_reconnect_on_socket_error(void)
+{
+	int err;
+	struct downloader_evt evt;
+
+	err = downloader_init(&dlc, &dlc_config);
+	TEST_ASSERT_EQUAL(0, err);
+
+	zsock_getaddrinfo_fake.custom_fake = zsock_getaddrinfo_server_ok;
+	zsock_freeaddrinfo_fake.custom_fake = zsock_freeaddrinfo_server_ipv6;
+	z_impl_zsock_socket_fake.custom_fake = z_impl_zsock_socket_http_ipv6_ok;
+	z_impl_zsock_connect_fake.custom_fake = z_impl_zsock_connect_ipv6_ok;
+	z_impl_zsock_setsockopt_fake.custom_fake = z_impl_zsock_setsockopt_http_ok;
+	z_impl_zsock_sendto_fake.custom_fake = z_impl_zsock_sendto_ok;
+	z_impl_zsock_recvfrom_fake.custom_fake =
+		z_impl_zsock_recvfrom_http_header_and_frag_data_w_err;
+
+	/* Default to http when no proto is specified. */
```

Comment:
This comment is not relevant to the code below. There are other occurrences of this.